src/jrd/trace/TraceConfigStorage.cpp

```diff
@@ -470,15 +470,20 @@ void ConfigStorage::compact()
 	ULONG check_used, check_size;
 	check_used = check_size = sizeof(TraceCSHeader);
 
+	// Track undeleted slots from dead storages
+	Firebird::NonPooledMap<ULONG, bool> deadStorages;
+
 	// collect used slots, sort them by offset
 	for (TraceCSHeader::Slot* slot = header->slots; slot < header->slots + header->slots_cnt; slot++)
 	{
 		if (slot->used && slot->ses_pid != pid &&
 			((slot->ses_flags & trs_system) == 0) && // System sessions are shared for multiple connections so they may live without the original process
 			!ISC_check_process_existence(slot->ses_pid))
 		{
-			fb_assert(header->cnt_uses != 0);
-			header->cnt_uses--; // Process that created trace session disappeared, count it out
+			// A SUPER server may shut down, but its Storage shared memory continues to live due to an embedded user session.
+			// The storage might allocate multiple slots, so count them carefully.
```

Comment:
Storage doesn't allocates slots, process does :)

Reply:
Corrected the description